ports/esp32/esp32_rmt.c

```diff
@@ -160,63 +160,107 @@ static mp_obj_t esp32_rmt_make_new(const mp_obj_type_t *type, size_t n_args, siz
 
     return MP_OBJ_FROM_PTR(self);
 }
 
 static void esp32_rmt_print(const mp_print_t *print, mp_obj_t self_in, mp_print_kind_t kind) {
     esp32_rmt_obj_t *self = MP_OBJ_TO_PTR(self_in);
     if (self->pin != -1) {
-        bool idle_output_en;
-        rmt_idle_level_t idle_level;
-        check_esp_err(rmt_get_idle_level(self->channel_id, &idle_output_en, &idle_level));
-        mp_printf(print, "RMT(channel=%u, pin=%u, source_freq=%u, clock_div=%u, idle_level=%u)",
-            self->channel_id, self->pin, APB_CLK_FREQ, self->clock_div, idle_level);
+        mp_printf(print, "RMT(pin=%u, source_freq=%u, resolution_hz=%u, idle_level=%u)",
+            self->pin, self->clock_freq, self->resolution_hz, self->idle_level);
     } else {
         mp_printf(print, "RMT()");
     }
 }
 
+static void esp32_rmt_deactivate(esp32_rmt_obj_t *self) {
+    if (self->enabled) {
+        // FIXME: panics in ESP32 if called while TX is ongoing and TX sequence is long (>300ms)
+        // Does not panic in ESP32-S3, ESP32-C3 and ESP32-C6.
+        // Tested with ESP-IDF up to 5.5
+        // ESP-IDF issue: https://github.com/espressif/esp-idf/issues/17692
+        //
+        // Cause is Interrupt WDT to trigger because ESP-IDF rmt_disable() disables
+        // interrupts and spinlocks until the ongoing TX sequence is finished.
+        //
+        // Workaround is never try to stop RMT sequences longer than 300ms (which are unusual
+        // anyway). Or apply the patch mentioned at the GitHub issue to ESP-IDF.
+        rmt_disable(self->channel);
+        self->enabled = false;
+    }
+}
+
+static mp_obj_t esp32_rmt_active(size_t n_args, const mp_obj_t *args) {
+    esp32_rmt_obj_t *self = MP_OBJ_TO_PTR(args[0]);
+
+    if (n_args == 1) {
+        return self->enabled ? mp_const_true : mp_const_false;
+    } else if (mp_obj_is_true(args[1])) {
+        mp_raise_ValueError(MP_ERROR_TEXT("Activate by calling write_pulses()"));
+    }
+
+    esp32_rmt_deactivate(self);
+
+    return mp_const_false;
+}
+static MP_DEFINE_CONST_FUN_OBJ_VAR_BETWEEN(esp32_rmt_active_obj, 1, 2, esp32_rmt_active);
+
 static mp_obj_t esp32_rmt_deinit(mp_obj_t self_in) {
-    // fixme: check for valid channel. Return exception if error occurs.
     esp32_rmt_obj_t *self = MP_OBJ_TO_PTR(self_in);
+
     if (self->pin != -1) { // Check if channel has already been deinitialised.
-        rmt_driver_uninstall(self->channel_id);
+        esp32_rmt_deactivate(self);
+        rmt_tx_event_callbacks_t callbacks = {
+            .on_trans_done = NULL,
+        };
+        rmt_tx_register_event_callbacks(self->channel, &callbacks, self);
+        rmt_del_encoder(self->encoder);
+        rmt_del_channel(self->channel);
         self->pin = -1; // -1 to indicate RMT is unused
+        self->tx_ongoing = 0;
         m_free(self->items);
+        return mp_const_true;
     }
-    return mp_const_none;
+
+    return mp_const_false;
```

Comment:
IMO deinit should return `None`.

Returning a bool allows you to distinguish between "RMT was deinit'd" and "RMT is already deinit'd".  But is that really needed?